pkg/util/test/util.go

```diff
@@ -1042,23 +1054,49 @@ func ConstructPxReleaseManifestURL(specGenURL string) (string, error) {
 	return u.String(), nil
 }
 
-func validateStorageClusterIsOnline(cluster *corev1.StorageCluster, timeout, interval time.Duration) (*corev1.StorageCluster, error) {
-	t := func() (interface{}, bool, error) {
+func validateStorageClusterInState(cluster *corev1.StorageCluster, status string) func() (interface{}, bool, error) {
+	return func() (interface{}, bool, error) {
 		cluster, err := operatorops.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
 		if err != nil {
 			return nil, true, fmt.Errorf("failed to get StorageCluster %s in %s, Err: %v", cluster.Name, cluster.Namespace, err)
 		}
 
-		if cluster.Status.Phase != "Online" {
+		if cluster.Status.Phase != status {
 			if cluster.Status.Phase == "" {
 				return nil, true, fmt.Errorf("failed to get cluster status")
 			}
 			return nil, true, fmt.Errorf("cluster state: %s", cluster.Status.Phase)
 		}
 		return cluster, false, nil
 	}
+}
+
+func validateAllStorageNodesInState(namespace string, status corev1.NodeConditionStatus) func() (interface{}, bool, error) {
```

Comment:
This validation does not co-relate with the test case you are running. Its not enough just to check if the storage node is online and say its passed. We need to get the configuration of the storage nodes and verify them after state is online on successful test.

Reply:
This is only run for testing cases where portworx should **not** start up successfully, because this is a more consistent way to find that installation failed (the operator `StorageCluster` status doesn't update, but the `StorageNode` status does). The success cases still go through the full validation path that the operator tests use before.

Reply:
Talked over with @vilasd in Zoom, we need to add functionality to check that at least the storage sections (or anything else relevant to the test) are set correctly in the storage cluster once it's applied. Should be relatively easy, I'm working on it now and will leave space for future improvements.